filters/include/pcl/filters/impl/covariance_sampling.hpp

```diff
@@ -62,20 +62,25 @@ pcl::CovarianceSampling<PointT, PointNT>::initCompute ()
   // Prepare the point cloud by centering at the origin and then scaling the points such that the average distance from
   // the origin is 1.0 => rotations and translations will have the same magnitude
   Eigen::Vector3f centroid (0.f, 0.f, 0.f);
-  for (std::size_t p_i = 0; p_i < indices_->size (); ++p_i)
-    centroid += (*input_)[(*indices_)[p_i]].getVector3fMap ();
-  centroid /= float (indices_->size ());
+  for (auto indit = indices_->begin(); indit != indices_->end(); ++indit) {
```

Comment:
`for (const auto& idx: *indices_) { centroid += (*input_)[idx].getVector3fMap(); }`

Reply:
Didn't mean to commit this (I was exploring the test failure), but I took your suggestion